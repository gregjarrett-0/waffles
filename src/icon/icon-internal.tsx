import React from 'react';

const sizeMap = {
  xsmall: 12,
  small: 14,
  medium: 16,
  large: 18,
  xlarge: 24,
};

type IconInternalProps = {
  /* Defines the size of the Icon. In general, use default size. */
  /* @default medium */
  size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
  /* [skip docs] */
  children: React.ReactNode;
} & React.SVGAttributes<SVGElement>;

function IconInternal({
  size = 'medium',
  children,
  ...restProps
}: IconInternalProps) {
  const numericSize = sizeMap[size];

  return (
    <svg
      width={numericSize}
      height={numericSize}
      aria-hidden="true"
      {...restProps}
    >
      {children}
    </svg>
  );
}

export default IconInternal;
