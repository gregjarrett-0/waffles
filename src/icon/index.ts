export { default as GoogleColoredBrand } from './generated/google-colored-brand';
export { default as PaypalColoredBrand } from './generated/paypal-colored-brand';
export { default as SlackColoredBrand } from './generated/slack-colored-brand';
export { default as AddCircle } from './generated/add-circle';
export { default as AddUser } from './generated/add-user';
export { default as AppleBrand } from './generated/apple-brand';
export { default as Apply } from './generated/apply';
export { default as Archive } from './generated/archive';
export { default as ArrowDown } from './generated/arrow-down';
export { default as ArrowLeft } from './generated/arrow-left';
export { default as ArrowRight } from './generated/arrow-right';
export { default as ArrowUp } from './generated/arrow-up';
export { default as Assess } from './generated/assess';
export { default as Assignment } from './generated/assignment';
export { default as Attachment } from './generated/attachment';
export { default as Attention } from './generated/attention';
export { default as Back } from './generated/back';
export { default as BarChart } from './generated/bar-chart';
export { default as Bell } from './generated/bell';
export { default as Book } from './generated/book';
export { default as Bookmark } from './generated/bookmark';
export { default as Branch } from './generated/branch';
export { default as Briefcase } from './generated/briefcase';
export { default as Building } from './generated/building';
export { default as Bulb } from './generated/bulb';
export { default as Bullet } from './generated/bullet';
export { default as Calendar } from './generated/calendar';
export { default as Camera } from './generated/camera';
export { default as Chat } from './generated/chat';
export { default as CheckmarkCircle } from './generated/checkmark-circle';
export { default as Checkmark } from './generated/checkmark';
export { default as ChevronDown } from './generated/chevron-down';
export { default as ChevronLeft } from './generated/chevron-left';
export { default as ChevronRight } from './generated/chevron-right';
export { default as ChevronUp } from './generated/chevron-up';
export { default as Circle } from './generated/circle';
export { default as Clock } from './generated/clock';
export { default as ClosedCaptions } from './generated/closed-captions';
export { default as Code } from './generated/code';
export { default as Cog } from './generated/cog';
export { default as Columns } from './generated/columns';
export { default as Command } from './generated/command';
export { default as Compass } from './generated/compass';
export { default as Copy } from './generated/copy';
export { default as CreditCard } from './generated/credit-card';
export { default as CrossCircle } from './generated/cross-circle';
export { default as Cross } from './generated/cross';
export { default as Crosshair } from './generated/crosshair';
export { default as DataCampBrand } from './generated/data-camp-brand';
export { default as DataCampCertificationBrand } from './generated/data-camp-certification-brand';
export { default as DataCampJobsBrand } from './generated/data-camp-jobs-brand';
export { default as DataCampLearnBrand } from './generated/data-camp-learn-brand';
export { default as DataCampRecruitBrand } from './generated/data-camp-recruit-brand';
export { default as DataCampWorkspaceBrand } from './generated/data-camp-workspace-brand';
export { default as Database } from './generated/database';
export { default as Desktop } from './generated/desktop';
export { default as Diamond } from './generated/diamond';
export { default as Document } from './generated/document';
export { default as Dollar } from './generated/dollar';
export { default as Download } from './generated/download';
export { default as Edit } from './generated/edit';
export { default as Education } from './generated/education';
export { default as Exercise } from './generated/exercise';
export { default as Exit } from './generated/exit';
export { default as Expand } from './generated/expand';
export { default as ExternalLink } from './generated/external-link';
export { default as FacebookBrand } from './generated/facebook-brand';
export { default as Feed } from './generated/feed';
export { default as FigmaBrand } from './generated/figma-brand';
export { default as Filter } from './generated/filter';
export { default as Flag } from './generated/flag';
export { default as Flash } from './generated/flash';
export { default as Flashcard } from './generated/flashcard';
export { default as Float } from './generated/float';
export { default as Folder } from './generated/folder';
export { default as Footer } from './generated/footer';
export { default as Forbidden } from './generated/forbidden';
export { default as Forward } from './generated/forward';
export { default as FullScreen } from './generated/full-screen';
export { default as GithubBrand } from './generated/github-brand';
export { default as Globe } from './generated/globe';
export { default as GoogleBrand } from './generated/google-brand';
export { default as GooglePlayBrand } from './generated/google-play-brand';
export { default as Group } from './generated/group';
export { default as Header } from './generated/header';
export { default as Health } from './generated/health';
export { default as Heart } from './generated/heart';
export { default as HelpCircle } from './generated/help-circle';
export { default as Help } from './generated/help';
export { default as Hidden } from './generated/hidden';
export { default as Home } from './generated/home';
export { default as Image } from './generated/image';
export { default as InfoCircle } from './generated/info-circle';
export { default as InstagramBrand } from './generated/instagram-brand';
export { default as Integration } from './generated/integration';
export { default as Interface } from './generated/interface';
export { default as Introduction } from './generated/introduction';
export { default as KaggleBrand } from './generated/kaggle-brand';
export { default as Keyboard } from './generated/keyboard';
export { default as Lab } from './generated/lab';
export { default as LeanpubBrand } from './generated/leanpub-brand';
export { default as Learn } from './generated/learn';
export { default as LineChart } from './generated/line-chart';
export { default as Link } from './generated/link';
export { default as LinkedInBrand } from './generated/linked-in-brand';
export { default as List } from './generated/list';
export { default as Locked } from './generated/locked';
export { default as Mail } from './generated/mail';
export { default as Markdown } from './generated/markdown';
export { default as Matrix } from './generated/matrix';
export { default as Medal } from './generated/medal';
export { default as Menu } from './generated/menu';
export { default as Microphone } from './generated/microphone';
export { default as Minimize } from './generated/minimize';
export { default as Minus } from './generated/minus';
export { default as Mobile } from './generated/mobile';
export { default as Moon } from './generated/moon';
export { default as More } from './generated/more';
export { default as Multidoc } from './generated/multidoc';
export { default as Mute } from './generated/mute';
export { default as NineDots } from './generated/nine-dots';
export { default as Pause } from './generated/pause';
export { default as PaypalBrand } from './generated/paypal-brand';
export { default as Phone } from './generated/phone';
export { default as PieChart } from './generated/pie-chart';
export { default as Pin } from './generated/pin';
export { default as PlatformSwitch } from './generated/platform-switch';
export { default as Play } from './generated/play';
export { default as Plus } from './generated/plus';
export { default as Practice } from './generated/practice';
export { default as Publish } from './generated/publish';
export { default as PythonBrand } from './generated/python-brand';
export { default as RBrand } from './generated/r-brand';
export { default as Random } from './generated/random';
export { default as Redo } from './generated/redo';
export { default as Refresh } from './generated/refresh';
export { default as Remove } from './generated/remove';
export { default as Reply } from './generated/reply';
export { default as Rewind } from './generated/rewind';
export { default as Rocket } from './generated/rocket';
export { default as Rows } from './generated/rows';
export { default as Save } from './generated/save';
export { default as ScalaBrand } from './generated/scala-brand';
export { default as Scissors } from './generated/scissors';
export { default as Search } from './generated/search';
export { default as Share } from './generated/share';
export { default as Skip } from './generated/skip';
export { default as SlackBrand } from './generated/slack-brand';
export { default as SortAscending } from './generated/sort-ascending';
export { default as SortDescending } from './generated/sort-descending';
export { default as Sort } from './generated/sort';
export { default as Sound } from './generated/sound';
export { default as Sparkles } from './generated/sparkles';
export { default as Star } from './generated/star';
export { default as Stop } from './generated/stop';
export { default as Sun } from './generated/sun';
export { default as Tab } from './generated/tab';
export { default as Tablet } from './generated/tablet';
export { default as Tag } from './generated/tag';
export { default as Team } from './generated/team';
export { default as Terminal } from './generated/terminal';
export { default as Text } from './generated/text';
export { default as Trash } from './generated/trash';
export { default as TrimBegin } from './generated/trim-begin';
export { default as TrimEnd } from './generated/trim-end';
export { default as Trophy } from './generated/trophy';
export { default as TwitterBrand } from './generated/twitter-brand';
export { default as Undo } from './generated/undo';
export { default as Unlink } from './generated/unlink';
export { default as Unlocked } from './generated/unlocked';
export { default as Upload } from './generated/upload';
export { default as Upvote } from './generated/upvote';
export { default as User } from './generated/user';
export { default as Verified } from './generated/verified';
export { default as VersionHistory } from './generated/version-history';
export { default as VideoCamera } from './generated/video-camera';
export { default as Video } from './generated/video';
export { default as Visible } from './generated/visible';
export { default as Waypoint } from './generated/waypoint';
export { default as Website } from './generated/website';
export { default as YoutubeBrand } from './generated/youtube-brand';
export { default as AddCircleSolid } from './generated/add-circle-solid';
export { default as AssignmentSolid } from './generated/assignment-solid';
export { default as AttentionSolid } from './generated/attention-solid';
export { default as BookmarkSolid } from './generated/bookmark-solid';
export { default as ChatSolid } from './generated/chat-solid';
export { default as CheckmarkCircleSolid } from './generated/checkmark-circle-solid';
export { default as CircleSolid } from './generated/circle-solid';
export { default as ClockSolid } from './generated/clock-solid';
export { default as CogSolid } from './generated/cog-solid';
export { default as CrossCircleSolid } from './generated/cross-circle-solid';
export { default as ExerciseSolid } from './generated/exercise-solid';
export { default as FlashSolid } from './generated/flash-solid';
export { default as GroupSolid } from './generated/group-solid';
export { default as HealthSolid } from './generated/health-solid';
export { default as HeartSolid } from './generated/heart-solid';
export { default as HelpCircleSolid } from './generated/help-circle-solid';
export { default as HiddenSolid } from './generated/hidden-solid';
export { default as InfoCircleSolid } from './generated/info-circle-solid';
export { default as LockedSolid } from './generated/locked-solid';
export { default as MicrophoneSolid } from './generated/microphone-solid';
export { default as MoonSolid } from './generated/moon-solid';
export { default as MuteSolid } from './generated/mute-solid';
export { default as PinSolid } from './generated/pin-solid';
export { default as PlaySolid } from './generated/play-solid';
export { default as RemoveSolid } from './generated/remove-solid';
export { default as RewindSolid } from './generated/rewind-solid';
export { default as RocketSolid } from './generated/rocket-solid';
export { default as SkipSolid } from './generated/skip-solid';
export { default as SoundSolid } from './generated/sound-solid';
export { default as SparklesSolid } from './generated/sparkles-solid';
export { default as StarSolid } from './generated/star-solid';
export { default as StopSolid } from './generated/stop-solid';
export { default as SunSolid } from './generated/sun-solid';
export { default as TerminalSolid } from './generated/terminal-solid';
export { default as TrashSolid } from './generated/trash-solid';
export { default as UnlockedSolid } from './generated/unlocked-solid';
export { default as UpvoteSolid } from './generated/upvote-solid';
export { default as UserSolid } from './generated/user-solid';
export { default as VerifiedSolid } from './generated/verified-solid';
export { default as VideoCameraSolid } from './generated/video-camera-solid';
export { default as VideoSolid } from './generated/video-solid';
export { default as VisibleSolid } from './generated/visible-solid';
