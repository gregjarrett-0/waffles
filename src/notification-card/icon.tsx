import {
  CheckmarkCircleInverted,
  CrossCircleInverted,
  AttentionInverted,
  InfoCircleInverted,
  RocketInverted,
} from '../icon';

import { iconWrapperStyle, iconStyle, iconBackgroundStyle } from './styles';
import NotificationCard from './notification-card';

type IconProps = {
  variant: NonNullable<
    React.ComponentProps<typeof NotificationCard>['variant']
  >;
  inverted: boolean;
};

function Icon({ variant, inverted }: IconProps) {
  function renderIcon() {
    switch (variant) {
      case 'success':
        return <CheckmarkCircleInverted />;
      case 'warning':
        return <AttentionInverted />;
      case 'error':
        return <CrossCircleInverted />;
      case 'upgrade':
        return <RocketInverted />;
      default:
        return <InfoCircleInverted />;
    }
  }

  return (
    <div css={iconWrapperStyle()}>
      <div css={iconBackgroundStyle({ variant, inverted })} />
      <div css={iconStyle({ variant, inverted })}>{renderIcon()}</div>
    </div>
  );
}

export default Icon;
