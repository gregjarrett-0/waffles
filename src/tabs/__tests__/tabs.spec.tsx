import React, { useState } from 'react';
import userEvent from '@testing-library/user-event';
import { render, fireEvent } from '@testing-library/react';

import { Tabs } from '../index';
import { AddCircle } from '../../icon';

const MOCKED_ID = '123abC';

jest.mock('nanoid', () => {
  return {
    nanoid: () => MOCKED_ID,
  };
});

type ControlledTabsProps = {
  onFirstTabFocus?: React.FocusEventHandler<HTMLButtonElement>;
  onFirstTabClick?: React.MouseEventHandler<HTMLButtonElement>;
};

function ControlledTabs({
  onFirstTabFocus,
  onFirstTabClick,
}: ControlledTabsProps) {
  const [activeTabIndex, setActiveTabIndex] = useState<React.Key>(0);

  return (
    <Tabs
      activeTab={activeTabIndex}
      onChange={(activeTab) => {
        setActiveTabIndex(activeTab);
      }}
    >
      <Tabs.Tab
        onFocus={onFirstTabFocus}
        onClick={onFirstTabClick}
        label="Taylor Tab"
        data-testid="taylor-tab"
      >
        Content of the first tab.
      </Tabs.Tab>
      <Tabs.Tab label="Ariana Tab" data-testid="ariana-tab">
        Content of the second tab.
      </Tabs.Tab>
      <Tabs.Tab label="Justin Tab" data-testid="justin-tab">
        Content of the third tab.
      </Tabs.Tab>
    </Tabs>
  );
}

function CustomKeysTabs() {
  const [activeTabIndex, setActiveTabIndex] = useState<React.Key>('ariana');

  return (
    <Tabs
      activeTab={activeTabIndex}
      onChange={(activeTab) => {
        setActiveTabIndex(activeTab);
      }}
    >
      <Tabs.Tab key="taylor" label="Taylor Tab" data-testid="taylor-tab">
        Content of the first tab.
      </Tabs.Tab>
      <Tabs.Tab key="ariana" label="Ariana Tab" data-testid="ariana-tab">
        Content of the second tab.
      </Tabs.Tab>
      <Tabs.Tab key="justin" label="Justin Tab" data-testid="justin-tab">
        Content of the third tab.
      </Tabs.Tab>
    </Tabs>
  );
}

function AutoActivateTabs() {
  const [activeTabIndex, setActiveTabIndex] = useState<React.Key>(0);

  return (
    <Tabs
      activeTab={activeTabIndex}
      onChange={(activeTab) => {
        setActiveTabIndex(activeTab);
      }}
      autoActivate
    >
      <Tabs.Tab label="Taylor Tab" data-testid="taylor-tab">
        Content of the first tab.
      </Tabs.Tab>
      <Tabs.Tab label="Ariana Tab" data-testid="ariana-tab">
        Content of the second tab.
      </Tabs.Tab>
      <Tabs.Tab label="Justin Tab" data-testid="justin-tab">
        Content of the third tab.
      </Tabs.Tab>
    </Tabs>
  );
}

describe('Tabs', () => {
  it('renders tabs', () => {
    const { getByRole, getAllByRole } = render(<ControlledTabs />);

    const tabList = getByRole('tablist');
    const tabs = getAllByRole('tab');

    expect(tabList).toBeInTheDocument();
    expect(tabs).toHaveLength(3);
  });

  it("after clicking a tab, it's content is displayed in appropriate tab panel", () => {
    const { getByTestId, getByText } = render(<ControlledTabs />);

    const tab = getByTestId('ariana-tab');
    fireEvent.click(tab);
    const tabContent = getByText('Content of the second tab.');

    expect(tabContent).toBeInTheDocument();
  });

  it('tabs and tab panels have appropriate a11y attributes', () => {
    const { getByTestId, getByText } = render(<ControlledTabs />);

    const inactiveTab = getByTestId('taylor-tab');
    const activeTab = getByTestId('justin-tab');
    fireEvent.click(activeTab);
    const activeTabPanel = getByText('Content of the third tab.');

    expect(inactiveTab).toHaveAttribute('tabindex', '-1');
    expect(inactiveTab).toHaveAttribute('aria-selected', 'false');
    expect(activeTab).toHaveAttribute('tabindex', '0');
    expect(activeTab).toHaveAttribute('aria-selected', 'true');
    expect(activeTab).toHaveAttribute('id', `tabs-${MOCKED_ID}-tab-2`);
    expect(activeTab).toHaveAttribute(
      'aria-controls',
      `tabs-${MOCKED_ID}-tab-panel-2`,
    );
    expect(activeTabPanel).toHaveAttribute('role', 'tabpanel');
    expect(activeTabPanel).toHaveAttribute(
      'aria-labelledby',
      `tabs-${MOCKED_ID}-tab-2`,
    );
  });

  it('tab could be disabled', () => {
    const { getByTestId } = render(
      <Tabs activeTab={0}>
        <Tabs.Tab label="Taylor Tab" data-testid="taylor-tab">
          Content of the first tab.
        </Tabs.Tab>
        <Tabs.Tab label="Disabled Ariana Tab" data-testid="ariana-tab" disabled>
          Content of the second tab.
        </Tabs.Tab>
      </Tabs>,
    );

    const disabledTab = getByTestId('ariana-tab');
    expect(disabledTab).toBeDisabled();
  });

  it('tab could be rendered as a link', () => {
    const { container } = render(
      <Tabs activeTab={0}>
        <Tabs.Tab as="a" href="http://taylorswift.com" label="Taylor Tab">
          Content of the first tab.
        </Tabs.Tab>
        <Tabs.Tab as="a" href="http://ariana-grande.io" label="Ariana Tab">
          Content of the second tab.
        </Tabs.Tab>
      </Tabs>,
    );

    const linkTab = container.querySelectorAll('a')[0];

    expect(linkTab).toBeInTheDocument();
    expect(linkTab).toHaveTextContent('Taylor Tab');
    expect(linkTab).toHaveAttribute('href', 'http://taylorswift.com');
  });

  it("an icon is rendered next to the tab's label", () => {
    const { container, getByTestId } = render(
      <Tabs activeTab={0}>
        <Tabs.Tab
          icon={<AddCircle />}
          label="Taylor Tab"
          data-testid="taylor-tab"
        >
          Content of the first tab.
        </Tabs.Tab>
        <Tabs.Tab label="Disabled Ariana Tab" data-testid="ariana-tab" disabled>
          Content of the second tab.
        </Tabs.Tab>
      </Tabs>,
    );

    const tab = getByTestId('taylor-tab');
    const icon = container.querySelector('svg');

    expect(tab).toBeInTheDocument();
    expect(tab).toContainElement(icon);
    expect(icon).toHaveAttribute('width', '12');
    expect(icon).toHaveAttribute('height', '12');
  });

  it('focus on tab element is handled correctly', () => {
    const handleFocus = jest.fn();
    render(<ControlledTabs onFirstTabFocus={handleFocus} />);

    userEvent.tab();

    expect(handleFocus).toHaveBeenCalledTimes(1);
  });

  it('click on tab element is handled correctly', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <ControlledTabs onFirstTabClick={handleClick} />,
    );

    const firstTab = getByTestId('taylor-tab');
    fireEvent.click(firstTab);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('custom key value could be used to index tabs instead of default number', () => {
    const { getByTestId, getByText } = render(<CustomKeysTabs />);

    const defaultTab = getByTestId('ariana-tab');

    expect(defaultTab).toHaveAttribute('aria-selected', 'true');

    const activeTab = getByTestId('justin-tab');
    fireEvent.click(activeTab);
    const activeTabPanel = getByText('Content of the third tab.');

    expect(activeTab).toHaveAttribute('aria-selected', 'true');
    expect(activeTabPanel).toBeInTheDocument();
  });

  it('if autoActivate prop is passed, when tab is focused with arrow key, the appropriate panel is shown', () => {
    const { getByText } = render(<AutoActivateTabs />);

    userEvent.tab(); // focus first tab
    userEvent.keyboard('{arrowright}'); // focus second tab
    userEvent.keyboard('{arrowright}'); // focus third tab
    const activeTabPanel = getByText('Content of the third tab.');

    expect(activeTabPanel).toBeInTheDocument();
  });

  it('it is possible to cycle through tabs when using right arrow key', () => {
    const { getByText } = render(<AutoActivateTabs />);

    userEvent.tab(); // focus first tab
    userEvent.keyboard('{arrowright}'); // focus second tab
    userEvent.keyboard('{arrowright}'); // focus the last, third tab
    userEvent.keyboard('{arrowright}'); // focus first tab again
    const activeTabPanel = getByText('Content of the first tab.');

    expect(activeTabPanel).toBeInTheDocument();
  });

  it('it is possible to cycle through tabs when using left arrow key', () => {
    const { getByText } = render(<AutoActivateTabs />);

    userEvent.tab(); // focus first tab
    userEvent.keyboard('{arrowright}'); // focus second tab
    userEvent.keyboard('{arrowleft}'); // focus first tab again
    userEvent.keyboard('{arrowleft}'); // focus the last, third tab
    const activeTabPanel = getByText('Content of the third tab.');

    expect(activeTabPanel).toBeInTheDocument();
  });

  it('the last tab is selected, after the End key is pressed', () => {
    const { getByText } = render(<AutoActivateTabs />);

    userEvent.tab(); // focus first tab
    userEvent.keyboard('{end}'); // focus the last tab
    const activeTabPanel = getByText('Content of the third tab.');

    expect(activeTabPanel).toBeInTheDocument();
  });

  it('the first tab is selected, after the Home key is pressed', () => {
    const { getByText } = render(<AutoActivateTabs />);

    userEvent.tab(); // focus first tab
    userEvent.keyboard('{end}'); // focus the last tab
    userEvent.keyboard('{home}'); // focus first tab
    const activeTabPanel = getByText('Content of the first tab.');

    expect(activeTabPanel).toBeInTheDocument();
  });

  it('renders snapshot', () => {
    const { getByRole } = render(
      <Tabs activeTab={0}>
        <Tabs.Tab label="First Tab">Content of the first tab.</Tabs.Tab>
        <Tabs.Tab label="Second Tab">Content of the second tab.</Tabs.Tab>
      </Tabs>,
    );

    const tabList = getByRole('tablist');

    expect(tabList).toMatchSnapshot();
  });

  it('renders snapshot of inverted', () => {
    const { getByRole } = render(
      <Tabs activeTab={0} inverted>
        <Tabs.Tab label="First Tab">Content of the first tab.</Tabs.Tab>
        <Tabs.Tab label="Second Tab">Content of the second tab.</Tabs.Tab>
      </Tabs>,
    );

    const tabList = getByRole('tablist');

    expect(tabList).toMatchSnapshot();
  });
});
