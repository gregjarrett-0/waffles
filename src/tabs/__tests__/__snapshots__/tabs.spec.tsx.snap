// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders snapshot 1`] = `
.emotion-0 {
  width: 100%;
  margin-bottom: 16px;
  -webkit-mask-image: none;
  mask-image: none;
  overflow-y: hidden;
  overflow-x: auto;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-0::-webkit-scrollbar {
  display: none;
  width: 0;
  height: 0;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 100%;
  padding-left: 0;
  padding-right: 64px;
  border-bottom: 2px solid rgba(5, 25, 45, 0.15);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: 0;
  border: 0;
  border-bottom: 2px solid transparent;
  padding-left: 0;
  padding-right: 0;
  padding-bottom: 10px;
  padding-top: 4px;
  margin-bottom: -2px;
  margin-right: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  color: #5D6A77;
}

.emotion-2:last-of-type {
  margin-right: 0;
}

.emotion-2:disabled {
  cursor: default;
  opacity: 0.6;
}

.emotion-2:not(:disabled) {
  color: #05192D;
  font-weight: 800;
  border-bottom-color: #05192D;
}

.emotion-3 {
  display: block;
  margin-bottom: -1px;
}

.emotion-4 {
  display: block;
  font-weight: 800;
  color: transparent;
  clip: rect(0 0 0 0);
  height: 1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  white-space: nowrap;
  word-wrap: normal;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: 0;
  border: 0;
  border-bottom: 2px solid transparent;
  padding-left: 0;
  padding-right: 0;
  padding-bottom: 10px;
  padding-top: 4px;
  margin-bottom: -2px;
  margin-right: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  color: #5D6A77;
}

.emotion-5:last-of-type {
  margin-right: 0;
}

.emotion-5:disabled {
  cursor: default;
  opacity: 0.6;
}

.emotion-5:hover:not(:disabled) {
  color: rgba(5, 25, 45, 0.8);
  border-bottom-color: rgba(5, 25, 45, 0.4);
}

<div
  aria-orientation="horizontal"
  class="emotion-0"
  role="tablist"
>
  <div
    class="emotion-1"
  >
    <button
      aria-controls="tabs-123abC-tab-panel-0"
      aria-selected="true"
      class="emotion-2"
      id="tabs-123abC-tab-0"
      role="tab"
      tabindex="0"
      type="button"
    >
      <span
        class="emotion-3"
      >
        First Tab
        <span
          aria-hidden="true"
          class="emotion-4"
        >
          First Tab
        </span>
      </span>
    </button>
    <button
      aria-controls="tabs-123abC-tab-panel-1"
      aria-selected="false"
      class="emotion-5"
      id="tabs-123abC-tab-1"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <span
        class="emotion-3"
      >
        Second Tab
        <span
          aria-hidden="true"
          class="emotion-4"
        >
          Second Tab
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`Tabs renders snapshot of inverted 1`] = `
.emotion-0 {
  width: 100%;
  margin-bottom: 16px;
  -webkit-mask-image: none;
  mask-image: none;
  overflow-y: hidden;
  overflow-x: auto;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-0::-webkit-scrollbar {
  display: none;
  width: 0;
  height: 0;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 100%;
  padding-left: 0;
  padding-right: 64px;
  border-bottom: 2px solid rgba(255, 255, 255, 0.15);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: 0;
  border: 0;
  border-bottom: 2px solid transparent;
  padding-left: 0;
  padding-right: 0;
  padding-bottom: 10px;
  padding-top: 4px;
  margin-bottom: -2px;
  margin-right: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  color: #9BA3AB;
}

.emotion-2:last-of-type {
  margin-right: 0;
}

.emotion-2:disabled {
  cursor: default;
  opacity: 0.6;
}

.emotion-2:not(:disabled) {
  color: #FFFFFF;
  font-weight: 800;
  border-bottom-color: #FFFFFF;
}

.emotion-3 {
  display: block;
  margin-bottom: -1px;
}

.emotion-4 {
  display: block;
  font-weight: 800;
  color: transparent;
  clip: rect(0 0 0 0);
  height: 1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  white-space: nowrap;
  word-wrap: normal;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: 0;
  border: 0;
  border-bottom: 2px solid transparent;
  padding-left: 0;
  padding-right: 0;
  padding-bottom: 10px;
  padding-top: 4px;
  margin-bottom: -2px;
  margin-right: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  transition: border-color 125ms ease-out,box-shadow 125ms ease-out;
  color: #9BA3AB;
}

.emotion-5:last-of-type {
  margin-right: 0;
}

.emotion-5:disabled {
  cursor: default;
  opacity: 0.6;
}

.emotion-5:hover:not(:disabled) {
  color: rgba(255, 255, 255, 0.8);
  border-bottom-color: rgba(255, 255, 255, 0.4);
}

<div
  aria-orientation="horizontal"
  class="emotion-0"
  role="tablist"
>
  <div
    class="emotion-1"
  >
    <button
      aria-controls="tabs-123abC-tab-panel-0"
      aria-selected="true"
      class="emotion-2"
      id="tabs-123abC-tab-0"
      role="tab"
      tabindex="0"
      type="button"
    >
      <span
        class="emotion-3"
      >
        First Tab
        <span
          aria-hidden="true"
          class="emotion-4"
        >
          First Tab
        </span>
      </span>
    </button>
    <button
      aria-controls="tabs-123abC-tab-panel-1"
      aria-selected="false"
      class="emotion-5"
      id="tabs-123abC-tab-1"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <span
        class="emotion-3"
      >
        Second Tab
        <span
          aria-hidden="true"
          class="emotion-4"
        >
          Second Tab
        </span>
      </span>
    </button>
  </div>
</div>
`;
